zkvm/src/blockchain/state.rs

```diff
@@ -62,6 +63,30 @@ impl BlockchainState {
 
         Ok(())
     }
+
+    /// Executes a list of transactions, returning their tx IDs and tx logs.
+    pub fn execute_txlist(
+        txs: Vec<Tx>,
+        bp_gens: BulletproofGens,
+        version: u64,
+        timestamp_ms: u64,
+    ) -> Result<Vec<(TxID, TxLog)>, BlockchainError> {
+        txs.par_iter()
+            .map(|tx| {
+                if tx.header.mintime_ms > timestamp_ms || tx.header.maxtime_ms < timestamp_ms {
+                    return Err(BlockchainError::BadTxTimestamp);
+                }
+                if version == 1 && version != tx.header.version {
```

Comment:
`version` here i guess is the block version, maybe rename it accordingly? This is very different from tx version, so comparison between them is semantically incorrect. The goal here is to say "if the block version is current, then tx version must be among currently supported to prevent squatting future versions; but if it's in the future, we are running an outdated software, so we need to allow higher tx versions."

Currently the set of supported block versions is simply {1} and the set of supported tx versions is also {1}, but those are ("type-wise") completely different 1's. It's conceivable that some other changes in the protocol would warrant bumping block versions to 2,3,4, but still keeping tx version capped at 1 for all of them.

Reply:
Did a name change and added a comment — lmk if that makes sense